extras/devices/LVFS/LVFS.py

```diff
@@ -0,0 +1,332 @@
+import subprocess
+import json
+import time
+from devices.base import AbstractDevice, logger
+
+SSH_OPTS = "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+
+class LVFSDevice(AbstractDevice):
+    fw_update_type = "LVFS"
+    vendor = ["HP", "Dell Inc.", "LENOVO"]
+    reboot_timeout = 600
+
+    def __init__(self, ipaddr, user, password):
+        super().__init__(ipaddr, user, password)
+        # self.run_cmd("sudo apt update && sudo apt install fwupd")
+
+    def run_cmd(self, cmd, raise_stderr=True, timeout=30):
+        """
+        Execute command on DUT via SSH
+
+        :returns: return code, stdout, stderr
+        :rtype: int, string, string
+        """
+        if self.password == "":
+            ssh_cmd = 'ssh -t %s %s@%s "%s"' % (
```

Comment:
Done.

Reply:
oops, I should have been more clear about NOT doing this in logging. I'll comment separately on that and close this one since you fixed all of the other ones.